prompt_toolkit/key_binding/bindings/emacs.py

```diff
@@ -234,6 +244,133 @@ def _(event: E) -> None:
         """
         event.current_buffer.exit_selection()
 
+    def unshift_move(event: E):
+        key = event.key_sequence[0].key
+        if key == Keys.ShiftLeft:
+            command = 'backward-char'
+        if key == Keys.ShiftRight:
+            command = 'forward-char'
+        if key == Keys.ShiftUp:
+            event.current_buffer.auto_up(count=event.arg)
+            return
+        if key == Keys.ShiftDown:
+            event.current_buffer.auto_down(count=event.arg)
+            return
+        if key == Keys.ShiftHome:
+            command = 'beginning-of-line'
+        if key == Keys.ShiftEnd:
+            command = 'end-of-line'
+        if key == Keys.ShiftControlLeft:
+            command = 'backward-word'
+        if key == Keys.ShiftControlRight:
+            command = 'forward-word'
+        if key == Keys.ShiftControlHome:
+            command = 'beginning-of-buffer'
+        if key == Keys.ShiftControlEnd:
+            command = 'end-of-buffer'
+        get_by_name(command)(event)
```

Comment:
Keep in mind that ``command`` is potentially undefined. This can raise a ``NameError``.

I think I'd prefer a lookup table that maps the keys to the commands. Something like this:
```python
if key == Keys.ShiftUp:
    ...
    return
...
key_to_command = {
     Keys.ShiftHome: 'beginning-of-line',
     Keys.ShiftEnd: 'end-of-line',
     ...
}
try:
    # Both the dict lookup and `get_by_name` can raise KeyError.
    handler = get_by_name(key_to_command[key])
except KeyError:
    pass
else:  # (`else` is not really needed here.)
    handler(event)
```